labrad/test/test_server.py

```diff
@@ -0,0 +1,32 @@
+import Queue
+
+import pytest
+
+import labrad
+from labrad.server import LabradServer, setting
+from labrad import util
+
+def test_server_expire_context_method_is_called():
+    """Ensure that server's expireContext method is called when client disconnects."""
+    queue = Queue.Queue()
+
+    class TestServer(LabradServer):
+        name = "TestServer"
+
+        def expireContext(self, c):
+            queue.put(c.ID)
+
+        @setting(100)
+        def echo(self, c, data):
+            return data
+
+    server = TestServer()
+    with util.syncRunServer(server):
+        with labrad.connect() as cxn:
+            context = (cxn.ID, 4)
+            cxn.testserver.echo('hello, world!', context=context)
+        expired = queue.get(block=True, timeout=1)
+        assert expired == context
```

Comment:
The `LabradServer` base class does some work behind the scenes to make context expiration work; in particular, it makes a request to the manager to indicate that it would like to receive context expiration messages, and it also registers a listener for those messages. Internally, it generates a context for that request to the manager, and matches the incoming messages against that context before calling the `expireContext` method. So the two contexts in play are: (1) the context `TestServer` used when registering with the manager to receive expiration messages, and (2) the context that is actually being expired, which is sent in the body of the expiration message.

I could add a note to this effect in the test, but I'm not sure it would be very illuminating since this all happens behind the scenes (there's no way to get at context (1) from this test code, for example).

As for the names, it's a good suggestion. Will change.